src/vis/density/HexagonalBin.tsx

```diff
@@ -0,0 +1,411 @@
+      setYZoomTransform(transform.y);
+    });
+
+    d3v7.select(`#${id}zoom`).call(zoom);
+  }, [id, xScale, yScale, height, width, config.dragMode]);
+
+  // apply brushing
+  useEffect(() => {
+    if (config.dragMode !== EScatterSelectSettings.RECTANGLE) {
+      d3v7.select(`#${id}brush`).selectAll('rect').remove();
+      return;
+    }
+
+    const brush = d3v7.brush().extent([
+      [margin.left, margin.top],
+      [margin.left + width, margin.top + height],
+    ]);
+    // it does look like we are creating a ton of brush events without cleaning them up right here.
+    // But d3v7.call will remove the previous brush event when called, so this actually works as expected.
+    d3v7.select(`#${id}brush`).call(
+      // this is a real function and not a => so that I can use d3v7.select(this) inside to clear the brush
+      brush.on('end', function (event: D3BrushEvent<any>) {
+        if (!event.sourceEvent) return;
+        if (!event.selection) {
+          selectionCallback([]);
+          return;
+        }
+
+        // To figure out if brushing is finding hexes after changing the axis via pan/zoom, need to do this.
+        // Invert your "zoomed" scale to find the actual scaled values inside of your svg coords. Use the original scale to find the values.
+        const startX = xZoomedScale.current ? xScale(xZoomedScale.current.invert(event.selection[0][0])) : event.selection[0][0];
+        const startY = yZoomedScale.current ? yScale(yZoomedScale.current.invert(event.selection[0][1])) : event.selection[0][1];
+        const endX = xZoomedScale.current ? xScale(xZoomedScale.current.invert(event.selection[1][0])) : event.selection[1][0];
+        const endY = yZoomedScale.current ? yScale(yZoomedScale.current.invert(event.selection[1][1])) : event.selection[1][1];
+
+        // to find the selected hexes
+        const selectedHexes = hexes.filter((currHex) =>
+          xZoomedScale.current
+            ? currHex.x >= startX && currHex.x <= endX && currHex.y >= startY && currHex.y <= endY
+            : currHex.x >= event.selection[0][0] &&
+              currHex.x <= event.selection[1][0] &&
+              currHex.y >= event.selection[0][1] &&
+              currHex.y <= event.selection[1][1],
+        );
+
+        const allSelectedPoints = selectedHexes.map((currHex) => currHex.map((points) => points[3])).flat();
+
+        selectionCallback(allSelectedPoints);
+
+        d3v7.select(this).call(brush.move, null);
+      }),
+    );
+  }, [width, height, id, hexes, selectionCallback, config.dragMode, xZoomTransform, yZoomTransform, zoomScale, xScale, yScale]);
+
+  // TODO: svg elements seem weird with style/classNames. I can directly put on a transform to a g, for example, but it seems to work
+  // differently than if i use style to do so
+  return (
+    <div ref={ref} className="mw-100">
+      <svg id={id} style={{ width: width + margin.left + margin.right, height: height + margin.top + margin.bottom }}>
```

Comment:
Setting the width and height of an SVG element is fine. Do you need to add the `px` unit for each of them if you use the `style`? Otherwise you could also use the `width` and `height` attribute, since they don't require a unit.